Update docs to reflect Azure 2022 process

Sieboldianus opened the pull request:
I recently went through the [Kubernetes on Microsoft Azure Kubernetes Service (AKS)](https://zero-to-jupyterhub.readthedocs.io/en/latest/kubernetes/microsoft/step-zero-azure.html) and [Kubernetes on Microsoft Azure Kubernetes Service (AKS) with Autoscaling](https://zero-to-jupyterhub.readthedocs.io/en/latest/kubernetes/microsoft/step-zero-azure-autoscale.html) docs. 

There were some minor inconsistencies with the current Azure process that I have updated.

Sieboldianus commented:
There's a broken link outside of the scope of this PR that is responsible for failed checks:
```
(administrator/upgrading/upgrade-1-to-2: line  114) broken    https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/2.0.0/images/hub/Dockerfile - 404 Client Error: Not Found for url: 
```

consideRatio commented:
@Sieboldianus wieeee thank you for helping out with this! I'm very positive on combining both the separate documents into one as that added significant complexity with regards to maintaining this documentation.

I'll try find time to review this tonight, but a key review point in my mind will be: was something of relevance lost when we removed one of the versions of the azure setup that should be retained in the one we keep?

Sieboldianus commented:
I can check again in detail, but it mostly looked duplicate, except for the preview-feature part (which isn't necessary anymore).

I'll also want to look into the part again for activating autoscale for an existing cluster, just to make sure it works as written.

Thank you for reviewing this. Thanks for the great docs!

Sieboldianus commented:
@consideRatio: I went through the docs again and compared the autoscaler to the standard docs and merged all.

- In the autoscaling docs, the paragraph `At this stage, you may wish to think about customising your deployment. The [Hub23 Deployment Guide](https://alan-turing-institute.github.io/hub23-deploy/) contains instructions for deploying a Kubernetes cluster to Azure with [autoscaling and multiple nodepools](https://alan-turing-institute.github.io/hub23-deploy/deploy-k8s/az-aks-create.html). These instructions can be combined so that all nodepools can autoscale.` was missing, but I guess this was just a result of diversion due to the duplicate handling. I left the paragraph as is.
- In the autoscaling docs, `--enable-vmss` now became `--vm-set-type VirtualMachineScaleSets`, the see: [#927](https://github.com/Azure/azure-cli-extensions/issues/927#issuecomment-534192033)
- I incorporated the part on adjusting autoscaler rules in the azure frontend to the standard docs:
    - Removed the part on "register for Microsoft Insights", since this is not necessary anymore
    - Updated the screenshots (I think it is also ok to stick with the old ones, the changes are minor, but I still created new ones)
- Verified all commands

I _think_ it makes sense merging both docs, since the differences between autoscaling and not autoscaling for a cluster are minimal, but I am new to the project and not the best to decide.

Sieboldianus commented:
While I was at it, I also updated [Installing JupyterHub: Getting the public IP of the Proxy](https://zero-to-jupyterhub.readthedocs.io/en/latest/jupyterhub/installation.html) ([dbd2f27](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2823/commits/dbd2f2782ca43712abae8b3da8342d980f0fb560))

`kubectl` outputs the recommended command, after installation of the helm chart:

Release "jupyterhub-eurocarto" does not exist. Installing it now.
NAME: jupyterhub-eurocarto
LAST DEPLOYED: Wed Aug  3 10:01:48 2022
NAMESPACE: eurocarto-namespace
> STATUS: deployed
> REVISION: 1
> TEST SUITE: None
> NOTES:
> Thank you for installing JupyterHub!
> 
> Your release is named "<helm-release-name>" and installed into the namespace "<k8s-namespace>".
> 
> You can check whether the hub and proxy are ready by running:
> 
>  kubectl --namespace=<k8s-namespace> get pod
> 
> and watching for both those pods to be in status 'Running'.
> 
> You can find the public (load-balancer) IP of JupyterHub by running:
> 
>   kubectl -n <k8s-namespace> get svc proxy-public -o jsonpath='{.status.loadBalancer.ingress[].ip}'
> 
> It might take a few minutes for it to appear!
> 
> To get full information about the JupyterHub proxy service run:
> 
>   kubectl --namespace=<k8s-namespace> get svc proxy-public
> 
> If you have questions, please:
> 
>   1. Read the guide at https://z2jh.jupyter.org
>   2. Ask for help or chat to us on https://discourse.jupyter.org/
>   3. If you find a bug please report it at https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues

This does not really fit in here. Happy to submit a different PR for this commit.

consideRatio commented on docs/source/jupyterhub/installation.md:
````diff
@@ -128,26 +128,23 @@ can try with `nano config.yaml`.
 
    ```
-   kubectl describe service proxy-public --namespace <k8s-namespace>
+   kubectl -n <k8s-namespace> get svc proxy-public -o jsonpath='{.status.loadBalancer.ingress[].ip}'
````
```suggestion
   kubectl --namespace <k8s-namespace> get service proxy-public --output jsonpath='{.status.loadBalancer.ingress[].ip}'
```

Going for small consistency tweak, and helping users avoid needing to guess that `svc` means `service` and -o means --output.

consideRatio replied:
```suggestion
   kubectl -namespace <k8s-namespace> get svc proxy-public --output jsonpath='{.status.loadBalancer.ingress[].ip}'
```

consideRatio commented on docs/source/jupyterhub/installation.md:
````diff
@@ -128,26 +128,23 @@ can try with `nano config.yaml`.
 
    ```
-   kubectl get service --namespace <k8s-namespace>
+   kubectl --namespace=<k8s-namespace> get svc proxy-public
````
```suggestion
   kubectl --namespace <k8s-namespace> get servuce proxy-public
```

consideRatio commented on docs/source/kubernetes/microsoft/step-zero-azure.md:
```diff
@@ -227,6 +230,46 @@ If you prefer to use the Azure portal see the [Azure Kubernetes Service quicksta
+      --query [0].name \
+      --output tsv)
+   az aks nodepool update \
+   --name $SP_POOLNAME \
+   --cluster-name <CLUSTER-NAME> \
+   --resource-group <RESOURCE-GROUP-NAME> \
+   --enable-cluster-autoscaler \
+   --min-count <DESIRED-MINIMUM-COUNT> \
+   --max-count <DESIRED-MAXIMUM-COUNT>
```
```suggestion
      --name $SP_POOLNAME \
      --cluster-name <CLUSTER-NAME> \
      --resource-group <RESOURCE-GROUP-NAME> \
      --enable-cluster-autoscaler \
      --min-count <DESIRED-MINIMUM-COUNT> \
      --max-count <DESIRED-MAXIMUM-COUNT>
```

consideRatio reviewed (approved):
Thank you @Sieboldianus !!!! :heart: :tada:
